dotnet/src/Connectors/Connectors.Amazon/Bedrock/Extensions/BedrockKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,575 @@
+
+        return builder;
+    }
+    /// <summary>
+    /// Add Amazon Bedrock Chat Completion service to the kernel builder using AWS Access Key ID and AWS Secret Access Key and
+    /// AWS Session Token and AmazonBedrockRuntimeClient Configuration Object.
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for chat completion.</param>
+    /// <param name="awsAccessKeyId">AWS Access Key ID</param>
+    /// <param name="awsSecretAccessKey">AWS Secret Access Key</param>
+    /// <param name="awsSessionToken">AWS Session Token</param>
+    /// <param name="clientConfig">The AmazonBedrockRuntimeClient Configuration Object</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockChatCompletionService(
+        this IKernelBuilder builder,
+        string modelId,
+        string awsAccessKeyId,
+        string awsSecretAccessKey,
+        string awsSessionToken,
+        AmazonBedrockRuntimeConfig clientConfig,
+        string? serviceId = null)
+    {
+        builder.Services.AddKeyedSingleton<IChatCompletionService>(serviceId, (_, _)
+            => new BedrockChatCompletionService(modelId, awsAccessKeyId, awsSecretAccessKey, awsSessionToken, clientConfig));
+
+        return builder;
+    }
+    /// <summary>
+    /// Add Amazon Bedrock Text Generation service to the kernel builder using IAmazonBedrockRuntime object.
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for text generation.</param>
+    /// <param name="bedrockApi">The IAmazonBedrockRuntime to run inference using the respective model.</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockTextGenerationService(
+        this IKernelBuilder builder,
+        string modelId,
+        IAmazonBedrockRuntime bedrockApi,
+        string? serviceId = null)
+    {
+        builder.Services.AddKeyedSingleton<ITextGenerationService>(serviceId, (_, _)
+            => new BedrockTextGenerationService(modelId, bedrockApi));
+
+        return builder;
+    }
+    /// <summary>
+    /// Add Amazon Bedrock Text Generation service to the kernel builder using new AmazonBedrockRuntimeClient().
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for text generation.</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockTextGenerationService(
+        this IKernelBuilder builder,
+        string modelId,
+        string? serviceId = null)
+    {
+        builder.Services.AddKeyedSingleton<ITextGenerationService>(serviceId, (_, _)
```

Comment:
Per my comment above about adding `IAmazonBedrockRuntimeClient` to the DI you would attempt to pull the `IAmazonBedrockRuntimeClient` from the `IServiceProvider` provided to the `Func`.